draft-ietf-oauth-selective-disclosure-jwt.md

```diff
@@ -385,128 +355,160 @@ plus a string suitable for computing a digest, but such approaches can easily le
+ * Create an array of three elements in this order:
+   1. A salt value. See (#salt-entropy) and (#salt_minlength) for security considerations. The salt value MUST be unique for each claim that is to be selectively disclosed. It is RECOMMENDED to base64url-encode the salt value, producing a string. Any other type that is allowed in JSON MAY be used, e.g., a number.
+   2. The claim name, or key, as it would be used in a regular JWT. This MUST be a string.
+   3. The claim's value, as it would be used in a regular JWT. The value MAY be of any type that is allowed in JSON, including numbers, strings, booleans, arrays, and objects.
+ * JSON-encode the array such that an UTF-8 string is produced.
+ * base64url-encode the byte representation of the UTF-8 string, producing a US-ASCII [@RFC0020] string. This string is the Disclosure.
 
-The following is an example for a JSON literal without Claim Name Blinding:
+The following example illustrates the steps described above.
 
-```
-{"s": "6qMQvRL5haj", "v": "Peter"}
+The array is created as follows:
+```json
+["6qMQvRL5haj", "family_name", "Möbius"]
 ```
 
-The following is an example for a JSON literal with Claim Name Blinding:
+The resulting Disclosure would be `WyI2cU1RdlJMNWhhaiIsICJmYW1pbHlfbmFtZSIsICJNw7ZiaXVzIl0`.
 
-```
-{"s": "6qMQvRL5haj", "v": "Peter", "n": "given_name"}
-```
+Note that the JSON encoding of the object is not canonicalized, so variations in white space, encoding
+of Unicode characters, and ordering of object properties are allowed. For example, the following strings
+are all valid and encode the same claim value:
+
+ * A different way to encode the umlaut: `WyI2cU1RdlJMNWhhaiIsICJmYW1pbHlfbmFtZSIsICJNXHUwMGY2Yml1cyJd`
+ * No white space: `WyI2cU1RdlJMNWhhaiIsImZhbWlseV9uYW1lIiwiTcO2Yml1cyJd`
+ * Newline characters between elements: `WwoiNnFNUXZSTDVoYWoiLAoiZmFtaWx5X25hbWUiLAoiTcO2Yml1cyIKXQ`
+
+#### Hashing Disclosures {#hashing_disclosures}
+
+For embedding the Disclosures in the SD-JWT, the Disclosures are hashed using the digest algorithm specified in the `sd_digest_derivation_alg` claim described below. The resulting hash is then included in the SD-JWT instead of the original claim value, as described next.
+
+The hash digest MUST be taken over the US-ASCII bytes of the base64url-encoded Disclosure. This follows the convention in JWS [@RFC7515] and JWE [@RFC7516]. The bytes of the hash digest MUST then be base64url-encoded.
+
+Note:
+
+ * The input to the hash function is the base64url-encoded Disclosure, not the bytes encoded by the base64url string.
+ * The bytes of the output of the hash function are base64url-encoded, not the bytes making up the (often used) hex representation of the bytes of the hash digest.
+
+For example, the
+SHA-256 hash digest of the Disclosure `WyI2cU1RdlJMNWhhaiIsICJmYW1pbHlfbmFtZSIsICJNw7ZiaXVzIl0` would be
+`uutlBuYeMDyjLLTpf6Jxi7yNkEF35jdyWMn9U7b_RYY`.
+
+#### Decoy Digests {#decoy_digests}
```

Comment:
I was also not familiar with a term decoy..

Reply:
It's a bit of a military term, but I hope it captures the spirit just fine.